tls/main.go

```diff
@@ -620,6 +637,24 @@ func osctrlService() {
 			time.Sleep(time.Duration(_t) * time.Second)
 		}
 	}()
+	if tlsConfig.MetricsEnabled {
+		log.Info().Msg("Metrics are enabled")
+		// Register Prometheus metrics
+		handlers.RegisterMetrics(prometheus.DefaultRegisterer)
+
+		// Creating a new prometheus service
+		prometheusServer := http.NewServeMux()
+		prometheusServer.Handle("/metrics", promhttp.Handler())
+
+		go func() {
+			log.Info().Msgf("Starting prometheus server at port %s", tlsConfig.MetricsPort)
+			err := http.ListenAndServe(":"+tlsConfig.MetricsPort, prometheusServer)
```

Comment:
I wonder if it makes sense to have an additional flag to provide here, otherwise the listener is going to be `0.0.0.0` by default, without any authentication.